src/pages/settings/Profile/PersonalDetails/DateOfBirthPage.js

```diff
@@ -38,6 +40,36 @@ class DateOfBirthPage extends Component {
 
         this.validate = this.validate.bind(this);
         this.updateDateOfBirth = this.updateDateOfBirth.bind(this);
+        this.getYearFromRouteParams = this.getYearFromRouteParams.bind(this);
+        this.minDate = moment().subtract(CONST.DATE_BIRTH.MAX_AGE, 'Y').toDate();
+        this.maxDate = moment().subtract(CONST.DATE_BIRTH.MIN_AGE, 'Y').toDate();
+
+        this.state = {
+            defaultYear: '',
+        };
+    }
+
+    componentDidMount() {
+        this.props.navigation.addListener('focus', this.getYearFromRouteParams);
+    }
+
+    componentWillUnmount() {
+        this.props.navigation.removeListener('focus', this.getYearFromRouteParams);
+    }
+
+    /**
+     * Function to be called to read year from params - necessary to read passed year from the Year picker which is a separate screen
+     * It allows to display selected year in the calendar picker without overwriting this value in Onyx
+     */
+    getYearFromRouteParams() {
+        const {params} = this.props.route;
+        if (params && params.year) {
+            this.setState({defaultYear: params.year});
+            this.datePicker.focus();
+            if (_.isFunction(this.datePicker.click)) {
+                this.datePicker.click();
+            }
```

Comment:
```suggestion
            if (this.datePicker) {
                this.datePicker.focus();
                if (_.isFunction(this.datePicker.click)) {
                    this.datePicker.click();
                }
            }
```

Reply:
```suggestion
            if (_.isFunction(this.datePicker.focus)) {
                    this.datePicker.focus();
                }
            if (_.isFunction(this.datePicker.click)) {
                this.datePicker.click();
            }
```
If we are checking if the function exists let's do that for both functions

Reply:
focus always exists, but click only exists on web.
on web focus will not show the date picker unlike native